ReactSkia/views/common/RSkTextUtils.cpp

```diff
@@ -0,0 +1,120 @@
+ /*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "ReactSkia/views/common/RSkTextUtils.h"
+
+using namespace skia::textlayout;
+
+namespace facebook {
+namespace react {
+namespace {
+inline int getLines(std::vector<LineMetrics>& metrics, float frameHeight) {
+    int minNumberOfLines = 0;
+    for(int i=0; i< metrics.size(); i++) {
+        if ((metrics[i]).fBaseline > frameHeight) {
+            minNumberOfLines = (metrics[i]).fLineNumber;
+            break;
+        }
+    }
+    return minNumberOfLines;
+}
+
+inline int getMaxNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLines, float frameHeight) {
+    int minNumberOfLines = getLines(metrics, frameHeight);
+    if ((minNumberOfLines) && (minNumberOfLines <= maxNumberOfLines))
+        return minNumberOfLines;
+        
+    return maxNumberOfLines;
+}
+
+inline int getTextLines(std::vector<LineMetrics> metrics, float maxNumberofLines, bool rectStrokeWidth, float frameHeight, float textFrameHeight) {
+    int numberOfLines = 0;
+
+    if (maxNumberofLines) {
+        numberOfLines = getMaxNumberOfLines(metrics, maxNumberofLines, frameHeight);
+    }
+    else if (rectStrokeWidth) {
+        numberOfLines = getLines(metrics, textFrameHeight);
+    }
+
+    return numberOfLines;
+}
+
+inline SkScalar yPosOffset(AttributedString attributedString, SkScalar paraHeight, Float frameHeight ) {
+    for(auto &fragment: attributedString.getFragments()) {
+        if((paraHeight < frameHeight) && (!fragment.textAttributes.textAlignVertical.empty())) {
+            if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"center"))
+                return ((frameHeight/2) - (paraHeight/2));
+            else if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"bottom"))
+                return (frameHeight - paraHeight);
+        }
+    }
+    return 0;
+}
+} //namespace
+
+namespace RSkTextUtils{
+void setText(std::shared_ptr<Paragraph>& paragraph,
```

Comment:
this API mostly is setting number of Lines. Can we modify the name as setTextLines ?